File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def ispalin(self,s,left,right):
        n = len(s)

        while left >= 0 and right < n:
            if s[left] != s[right]:
                break
            left -=1
            right += 1

        return s[left+1:right]

        

    def longestPalindrome(self, s: str) -> str:
        res = ''
        max_palin = 0

        for i in range(len(s)):
            odd_palin = self.ispalin(s,i,i)
            if odd_palin:
                if len(odd_palin) > max_palin:
                    max_palin = len(odd_palin)
                    res = odd_palin
            
            even_palin = self.ispalin(s,i,i+1)
            if even_palin:
                if len(even_palin) > max_palin:
                    max_palin = len(even_palin)
                    res = even_palin
        return res
                    

        


        

File: 5-longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        if(s.length() == 0) {
            return "";
        }
        if(s.length() == 1) {
            return s;
        }
        int max = Integer.MIN_VALUE;
        String res = "";
        for(int i=0; i<s.length()-1; i++) {
            for(int j=i; j<s.length(); j++) {
                if(isPaling(s.substring(i,j+1))) {

                    if(s.substring(i,j+1).length() > max) {
                        max = s.substring(i,j+1).length();
                        res = s.substring(i,j+1);
                    }
                }
            }
         }

    return res;
    }


    public boolean isPaling(String s) {
        int left = 0;
        int right = s.length()-1;

        while(left <= right) {
            if(s.charAt(left) != s.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }

        return true;
    }
}